src/Chapter3.hs

```diff
@@ -460,6 +514,24 @@ Create a simple enumeration for the meal types (e.g. breakfast). The one who
 comes up with the most number of names wins the challenge. Use your creativity!
 -}
 
+data Breakfast
+    = FrenchBreakfast
+        { croissant :: Int
+        , jam       :: Int
+        , butter    :: Int
+        }
+    | EnglishBreakfast
+        { tea       :: Int
+        , milk      :: Int
+        , friedEggs :: Int
+        , toasts    :: Int
+        }
+    | RussianBreakfast
+        { caviar    :: Int
+        , pancakes  :: Int
+        }
```

Comment:
Yummy 😋 🥞 Very creative 🎨 

Nice usage of both sums and record types!

One note in here, record inside the sum type is considered an antipattern in Haskell. This is because it produce partial functions.
As we mentioned, records bring filed names as function in scope.
In case of record, for example, we would have the function:

```haskell
jam :: Breakfast -> Int
```

However, this function is partial, as it will fail in the runtime on such input:

```haskell
jam (RussianBreakfast 1 2)
```

The solution in here is to more record into separate data type and use it inside the sum type as unnamed arg:

```haskell
data FrenchBreakfast = FrenchBreakfast
    { croissant :: Int    
    , jam       :: Int
    , butter    :: Int
    }
...

data Breakfast
    = French FrenchBreakfast
    | English EnglishBreakfast
    ...
```